src/ott/geometry/costs.py

```diff
@@ -327,7 +329,107 @@ def tree_unflatten(cls, aux_data, children):  # noqa: D102
     return cls(*aux_data)
 
 
-@jax.tree_util.register_pytree_node_class
+@jtu.register_pytree_node_class
+class RegTICost(TICost):
+  r"""Regularized translation-invariant cost.
+
+  .. math:
+    \frac{\rho}{2}\|\cdot\|_2^2 + \text{regularizer}(\cdot)
+
+  Args:
+    regularizer: Regularizer function.
+    rho: Scaling factor.
+  """
+
+  def __init__(
+      self, regularizer: regularizers.ProximalOperator, rho: float = 1.0
+  ):
+    self.regularizer = regularizer
+    self.rho = rho
+    self._h = regularizers.Regularization(
+        regularizer,
+        a=None,
+        rho=rho,
+    )
+
+  def h(self, z: jnp.ndarray) -> float:  # noqa: D102
+    return self._h(z)
+
+  def h_legendre(self, z: jnp.ndarray) -> float:  # noqa: D102
+
+    @jax.custom_vjp
+    def fn(z: jnp.ndarray) -> float:
+      out, _ = fwd(z)
+      return out
+
+    def fwd(z: jnp.ndarray) -> Tuple[float, jnp.ndarray]:
+      q = self.regularizer.prox(z)
+      return jnp.dot(q, z) - self.h(q), q
+
+    def bwd(q: jnp.ndarray, g: jnp.ndarray) -> Tuple[jnp.ndarray]:
+      return jnp.dot(g, q),
+
+    fn.defvjp(fwd, bwd)
+    return fn(z)
+
+  def h_transform(
+      self,
+      f: Callable[[jnp.ndarray], float],
+      x_init: Optional[jnp.ndarray] = None,
+      **kwargs: Any
+  ) -> Callable[[jnp.ndarray], float]:
+    r"""Compute the h-transform of a concave function.
+
+    Return a callable :math:`f_h` defined as:
+
+    .. math::
+      f_h(x) = \min_y h(x - y) - f(y)
+
+    This is equivalent, up to a change of variables, :math:`z = x - y`, to
+    define
+
+    .. math::
+      \min_z h(z) - f(x - z). \\
+      \min_z h(z) + \tilde{f}(z, x).
+
+    where :math:`\tilde{f}(z, x) := -f(x - z)`.
+
+    This is solved using proximal gradient descent, which requires having
+    access to the prox of :math:`\text{scaling_h} \cdot h` and not only to that
+    of :meth:`h`. Given the properties of :meth:`h`, the prox is obtained by
+    rescaling the output of the prox of a suitable scaling.
+
+    Args:
+      f: Concave function.
+      x_init: Initial estimate. If :obj:`None`, use ``x``.
+      kwargs: Keyword arguments for :class:`~jaxopt.ProximalGradient`.
+
+    Returns:
+      The h-transform of ``f``.
+    """
+
+    def f_h(x: jnp.ndarray) -> float:
+      solver = jaxopt.ProximalGradient(
+          fun=lambda z, x: -f(x - z),
+          prox=lambda x, reg, tau: reg.prox(x, tau),
```

Comment:
consider renaming reg, because its actually h

Reply:
also maybe tau is eta?

Reply:
> consider renaming reg, because its actually h

Thanks, done!

> also maybe tau is eta?

I think $\tau$ should be ok, to keep consistent what we had previously.